apps/balancer-tools/src/app/apr/api/(utils)/date.ts

```diff
@@ -48,3 +48,17 @@ export function getWeeksBetweenDates(
 
 export const dateToEpoch = (date: Date): number =>
   Math.floor(date.getTime() / 1000);
+
+export function isTimePeriodsIntersecting(
+  periodOneStartDateTimestamp: number,
+  periodOneEndDateTimestamp: number,
+  periodTwoStartDateTimestamp: number,
+  periodTwoEndDateTimestamp: number,
+): boolean {
+  return (
+    (periodTwoStartDateTimestamp <= periodOneEndDateTimestamp &&
+      periodTwoEndDateTimestamp >= periodOneStartDateTimestamp) ||
+    (periodOneStartDateTimestamp <= periodTwoEndDateTimestamp &&
+      periodOneEndDateTimestamp >= periodTwoStartDateTimestamp)
+  );
+}
```

Comment:
just a fancier implementation 😅 
```suggestion
type Timestamp = number;

export function doIntervalsIntersect(
  periodOneStart: Timestamp,
  periodOneEnd: Timestamp,
  periodTwoStart: Timestamp,
  periodTwoEnd: Timestamp
): boolean {
  return !((periodOneStart > periodTwoEnd) ^ (periodTwoStart > periodOneEnd));
}
```

Reply:
thanks!